core/node/base_token_adjuster/src/metrics.rs

```diff
@@ -0,0 +1,34 @@
+use std::time::Duration;
+
+use vise::{Buckets, EncodeLabelSet, EncodeLabelValue, Family, Histogram, Metrics};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, EncodeLabelSet, EncodeLabelValue)]
+#[metrics(label = "operation_result", rename_all = "snake_case")]
+pub(super) enum OperationResult {
+    Success,
+    Failure,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, EncodeLabelSet)]
+pub(crate) struct OperationResultLabels {
+    pub result: OperationResult,
+    pub attempts: u32,
+}
+
+/// Roughly exponential buckets for gas (10k – 50M).
+const GAS_BUCKETS: Buckets =
+    Buckets::values(&[1e4, 2e4, 5e4, 1e5, 2e5, 5e5, 1e6, 2e6, 5e6, 1e7, 2e7, 5e7]);
+
+#[derive(Debug, Metrics)]
+#[metrics(prefix = "snapshots_creator")]
+pub(crate) struct BaseTokenAdjusterMetrics {
+    #[metrics(buckets = GAS_BUCKETS)]
+    pub l1_gas_used: Histogram<f64>,
+    #[metrics(buckets = Buckets::LATENCIES)]
+    pub external_price_api_latency: Family<OperationResultLabels, Histogram<Duration>>,
+    #[metrics(buckets = Buckets::LATENCIES)]
+    pub l1_update_latency: Family<OperationResultLabels, Histogram<Duration>>,
```

Comment:
Maybe a nit, but I'm not sure it will be nice to show/interpret this data in grafana. We'll have different labels per attempt. This would make it easier to answer questions like "how much time does it take to update L1 **on the second attempt**? **On the third attempt**? I don't think these are the questions we are interested in.

We are interested in "how often does the request result in a error" and maybe what attempt is normally successful. So I'd just remove `attempts` from `OperationResultLabels`, report the counter for errors and maybe a gauge with the successful attempts

Reply:
Good point. Removing attempts. Will report just latency + label of "success" / "failure" for every attempt. Then we can sum up and plot those as we need.